src/test/java/jenkins/plugins/github/api/mock/MockGitHub.java

```diff
@@ -0,0 +1,233 @@
+                i0 = i1 + 1;
+                i1 = fragment.indexOf('/', i0);
+            }
+        }
+        return result.toString();
+    }
+
+    public String json(Object object) throws IOException {
+        if (object == null) {
+            return "null";
+        }
+        StringWriter w = new StringWriter();
+        JsonGenerator generator = factory.createGenerator(w);
+        generator.writeObject(object);
+        generator.close();
+        return w.toString();
+    }
+
+    public MockUser withUser(String login) {
+        MockUser result = new MockUser(this, login);
+        users.put(login, result);
+        return result;
+    }
+
+    public MockOrganization withOrg(String login) {
+        MockOrganization result = new MockOrganization(this, login);
+        organizations.put(login, result);
+        return result;
+    }
+
+    public List<MockOwner<?>> owners() {
+        List<MockOwner<?>> result = new ArrayList<>(organizations.size() + users.size());
+        result.addAll(users.values());
+        result.addAll(organizations.values());
+        return result;
+    }
+
+    public HttpResponse doRepositories(final @QueryParameter long since) {
+        return new HttpResponse() {
+            @Override
+            public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object node)
+                    throws IOException, ServletException {
+                List<MockRepository> repositories = new ArrayList<>();
+                for (MockOwner<?> o : owners()) {
+                    for (MockRepository r: o.repositories().values()) {
+                        if (r.getId() > since && !r.isPrivate()) {
+                            repositories.add(r);
+                        }
+                    }
+                }
+                Collections.sort(repositories, new Comparator<MockRepository>() {
+                    @Override
+                    public int compare(MockRepository o1, MockRepository o2) {
+                        return Long.compare(o1.getId(), o2.getId());
+                    }
+                });
+                if (repositories.size() > 30) {
+                    rsp.addHeader("Link", String.format(
+                            "<%s/repositories?since=%d>; rel=\"next\", <%s/repositories{?since}>; rel=\"first\"",
+                            getUrl(), repositories.get(30).getId(), getUrl()));
+                } else {
+                    rsp.addHeader("Link", String.format("<%s/repositories{?since}>; rel=\"first\"", getUrl()));
+                }
+                rsp.setContentType("application/json; charset=utf-8");
+                JsonGenerator o = factory.createGenerator(rsp.getOutputStream());
+                o.writeStartArray();
+                try {
+                    for (MockRepository r : repositories.subList(0, Math.min(30, repositories.size()))) {
+                        o.writeStartObject();
```

Comment:
Maybe I've missed something skimming through here but is there a reason we're doing manual mapping here rather than relying on Jackson serialization of the object type (with annotations as needed)?